eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/DependencyAnalyzer.java

```diff
@@ -0,0 +1,207 @@
+                    }
+//                    } else {
+//                        // If this is not a core dependency, we pick the highest version with the largest size.
+//                        int largestSize = 0;
+//                        String version = null;
+//
+//                        for (String orderedVersion : versionList) {
+//                            int currentSize = versionToDependency.get(orderedVersion).size();
+//                            if (currentSize > largestSize) {
+//                                largestSize = currentSize;
+//                                version = orderedVersion;
+//                            }
+//                        }
+//
+//                        BomDependency dependency = new BomDependency(key.getGroupId(), key.getArtifactId(), version);
+//                        bomEligibleDependencies.add(dependency);
+//                          logger.info("Multiple version of the dependency {} included.", key);
+//                          logger.info("\tPicking the highest version with largest dependents: {}", version);
+//
+//                        // Put all the other dependencies into the ineligible list.
+//                        for (String orderedVersion : versionList) {
+//                            if (orderedVersion != version) {
+//                                bomIneligibleDependencies.addAll(versionToDependency.get(orderedVersion));
+//                                  logger.info("\tDiscarding version:{}", orderedVersion);
+//                                  logger.info("\t\tDiscarding child dependencies {}", versionToDependency.get(orderedVersion).toString());
+//                            }
+//                        }
+//                    }
+                }
+            });
+
+        // There may still be some intersection between eligible and non eligible dependencies.
+        bomEligibleDependencies.removeAll(bomIneligibleDependencies);
+
+        return hasConflict.get();
+    }
+
+    private boolean filterConflicts() {
+        boolean hasconflict = resolveConflicts();
+        nameToVersionToChildrenDependencyTree.keySet().stream().forEach(
+            key -> {
+                HashMap<String, Collection<BomDependency>> versionToDependency = nameToVersionToChildrenDependencyTree.get(key);
+
+                if (versionToDependency.size() == 1) {
+                    BomDependency dependency = new BomDependency(key.getGroupId(), key.getArtifactId(), versionToDependency.keySet().stream().findFirst().get());
+                    if (!bomIneligibleDependencies.contains(dependency)
+                        && !externalDependencies.contains(dependency)) {
+                        // No conflict, the library can be added to the list.
+                        bomEligibleDependencies.add(dependency);
+                    }
+                }
+            });
```

Comment:
Because we may\may nor want to use the value of that analysis. 

The generator tool tries to find the right set and in order to do that it picks a bunch of dependencies and analyses the ones that confirms to the standard, drops the one that don't and then validates whether the ones it found are the right ones.

So, the result might still be useful to us at the time of conflict.